src/main/java/eu/europa/ted/eforms/sdk/analysis/SdkAnalyzer.java

```diff
@@ -2,32 +2,70 @@
 
 import java.io.IOException;
 import java.nio.file.Path;
+import java.util.stream.Stream;
 import javax.xml.bind.JAXBException;
 import javax.xml.parsers.ParserConfigurationException;
 import javax.xml.xpath.XPathExpressionException;
+import org.apache.commons.lang3.ArrayUtils;
 import org.apache.commons.lang3.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.xml.sax.SAXException;
 import eu.europa.ted.eforms.sdk.analysis.drools.RulesRunner;
 import eu.europa.ted.eforms.sdk.analysis.drools.SdkUnit;
+import eu.europa.ted.eforms.sdk.analysis.vo.SdkMetadata;
 import eu.europa.ted.eforms.sdk.util.SdkMetadataParser;
 
 public class SdkAnalyzer {
   private static final Logger logger = LoggerFactory.getLogger(SdkAnalyzer.class);
 
   private SdkAnalyzer() {}
 
-  public static int analyze(Path sdkRoot) throws IOException, JAXBException, SAXException,
-      ParserConfigurationException, XPathExpressionException {
+  public static int analyze(final Path sdkRoot) throws XPathExpressionException, IOException,
+      JAXBException, SAXException, ParserConfigurationException {
     logger.info("Analyzing SDK under folder [{}]", sdkRoot);
 
-    FactsLoader factsLoader = new FactsLoader(sdkRoot);
+    final SdkMetadata sdkMetadata = SdkMetadataParser.loadSdkMetadata(sdkRoot);
+
+    final Validator templatesValidator = analyzeTemplates(sdkRoot, sdkMetadata.getVersion());
+
+    final Validator sdkValidator = analyzeSdk(sdkRoot, sdkMetadata);
+
+    final String[] warnings =
+        concatArrays(templatesValidator.getWarnings(), sdkValidator.getWarnings());
+    final String[] errors = concatArrays(templatesValidator.getErrors(), sdkValidator.getErrors());
+
+    if (ArrayUtils.isNotEmpty(warnings) && logger.isWarnEnabled()) {
+      logger.warn("Validation warnings:\n{}", StringUtils.join(errors, '\n'));
+    }
+
+    if (ArrayUtils.isNotEmpty(errors) && logger.isWarnEnabled()) {
```

Comment:
if (ArrayUtils.isNotEmpty(errors) && logger.isWarnEnabled()) 
    isErrorEnabled should be used here

Reply:
Fixed